dcicutils/glacier_utils.py

```diff
@@ -0,0 +1,321 @@
+            versions = response.get('Versions', [])
+            versions.sort(key=lambda x: x['VersionId'], reverse=True)  # most recent version will be first
+
+            # Delete all glaciered versions if the flag is set
+            if delete_all_versions:
+                glacier_versions = [v for v in versions if v.get('StorageClass') in GLACIER_CLASSES]
+                for v in glacier_versions:
+                    response = self.s3.delete_object(
+                        Bucket=bucket,
+                        Key=key,
+                        VersionId=v.get('VersionId')
+                    )
+                    PRINT(f'Object {key} Glacier version {v.get("VersionId")} deleted:\n{response}')
+                # no Glacier versions were found
+                if not glacier_versions:
+                    PRINT(f'No Glacier versions found for object {bucket}/{key}')
+                    return False
+                else:
+                    return True
+            else:
+                # Find the first glacierized version and delete it
+                for v in versions:
+                    if v.get('StorageClass') in GLACIER_CLASSES:
+                        response = self.s3.delete_object(
+                            Bucket=bucket,
+                            Key=key,
+                            VersionId=v.get('VersionId')
+                        )
+                        PRINT(f'Object {key} Glacier version {v.get("VersionId")} deleted:\n{response}')
+                        break
+                else:
+                    PRINT(f'No Glacier version found for object {key}')
+
+        except Exception as e:
+            PRINT(f'Error deleting Glacier versions of object {key}: {str(e)}')
+            return False
+
+    def copy_object_back_to_original_location(self, bucket: str, key: str,
+                                              version_id: Union[str, None] = None) -> Union[dict, None]:
+        """ Reads the temporary location from the restored object and copies it back to the original location
+
+        :param bucket: bucket where object is stored
+        :param key: key within bucket where object is stored
+        :param version_id: version of object, if applicable
+        :return: boolean whether the copy was successful
+        """
+        try:
+            # Get temporary location
+            restore = self.extract_temporary_s3_location_from_restore_response(bucket, key)
+            # Copy the object from the temporary location to its original location
+            copy_source = {'Bucket': bucket, 'Key': key, 'Restore': restore}
+            copy_args = {'Bucket': bucket, 'Key': key}
+            if version_id:
+                copy_source['VersionId'] = version_id
+                copy_args['CopySourceVersionId'] = version_id
+            response = self.s3.copy_object(CopySource=copy_source, **copy_args)
+            PRINT(f'Response from boto3 copy:\n{response}')
```

Comment:
I would like this to show up both when a user is running it from the console and when run programmatically say within foursight. I _think_ PRINT accomplishes both?

Reply:
Well, it's just doing I/O to stdout by default (you can override with `file=` as in `print`), so yeah, if that's the normal way that it's done for other `foursight` things, I guess that's right. OK, just wanted to check.

That kind of gets to another question I had, which was who was the likely caller. These are potentially long-running things, so I assume it would not work well to do this kind of thing in a web request, for example.